### PR TITLE
Release 1.10.0 - add AWS SES option

### PR DESCRIPTION
**Pull request by briskt**
### Added
- Added the option to use AWS SES for sending email messages.

### Fixed
- Refactor: in `Email::send` use the return value instead of a specialized exception

### CODE REVIEW

File: application/common/components/SesMessage.php [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+namespace common\components;
+
+
+use yii\mail\BaseMessage;
+use yii\mail\MessageInterface;
+
+class SesMessage extends BaseMessage
+{
+    /** @var string */
+    private $_charset;
```
Comment (mtompset) [inline application/common/components/SesMessage.php:13]: Is it possible to strong type and remove the comment?

File: application/common/components/SesMessage.php [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+    public function setReplyTo($replyTo)
+    {
+        if (is_array($replyTo)) {
+            $this->_replyTo = $replyTo;
+        } else {
+            $this->_replyTo = explode(",", $replyTo);
+        }
+    }
+
+    public function getCc()
```
Comment (mtompset) [inline application/common/components/SesMessage.php:97]: Include PHPDoc on things that return array, so you know the contents of the array.

File: application/common/components/SesMessage.php [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+            $this->_cc = explode(",", $cc);
+        }
+    }
+
+    public function getBcc()
+    {
+        return $this->_bcc ?? [];
+    }
+
+    public function setBcc($bcc)
```
Comment (mtompset) [inline application/common/components/SesMessage.php:116]: void is your friend for set functions. :)

File: application/common/config/main.php [hunk 4]
```diff
@@ -1,22 +1,20 @@
 <?php
 
+use common\components\SesMailer;
 use Sil\JsonLog\target\JsonStreamTarget;
 use Sil\Log\EmailTarget;
 use Sil\PhpEnv\Env;
 use yii\db\Connection;
 use yii\helpers\Json;
-use yii\swiftmailer\Mailer;
+use yii\swiftmailer\Mailer as SwiftMailer;
```
Comment (mtompset) [inline application/common/config/main.php:9]: Swiftmailer is deprecated. So, you might consider refactoring this to use symfony.

File: application/tests/unit/common/models/EmailTest.php [hunk 5]
```diff
@@ -171,7 +171,8 @@ public function testSend()
 
         $this->assertEquals($initialEmailQueueCount + 1, Email::find()->count(), 'emails in db did not increase by one after saving email');
 
-        $email->send();
+        $n = $email->send();
+        $this->assertEquals(1, $n, 'message not sent');
```
Comment (mtompset) [inline application/tests/unit/common/models/EmailTest.php:175]: If you check, I'm pretty sure it's static. The code works, but it isn't the best.

File: docker-compose.yml [hunk 6]
```diff
@@ -20,6 +20,13 @@ services:
             - "8888:80"
         depends_on:
             - db
+        environment:
+            APP_NAME: email-service
```
Comment (mtompset) [inline docker-compose.yml:24]: I'd prefer sticking this in a .env file with a .env-dist example file. Don't want passwords leaking into source repositories. :)

File: application/common/config/main.php (hunk 4 above)
Comment (briskt) [reply application/common/config/main.php:9]: Added this as a new ticket (IDP-785).

File: application/tests/unit/common/models/EmailTest.php (hunk 5 above)
Comment (briskt) [reply application/tests/unit/common/models/EmailTest.php:175]: I don't know, and it's like this throughout this repo. I'd rather not risk screwing it up. Thanks for the suggestion, though.

File: docker-compose.yml (hunk 6 above)
Comment (briskt) [reply docker-compose.yml:24]: I'm assuming you mean `API_ACCESS_KEYS: abc123` even though the comment is tagged on `APP_NAME: email-service`. It's already there, but I'll go ahead and add the `abc123` to the dist file.

Comment (mtompset) [reply docker-compose.yml:24]: Actually, I meant all of them... even the MYSQL_ stuff below.

**Review by mtompset — APPROVED** [review]
Stale releases suck, so approved. I'm sure you can fix any problems that come up. :)

**Pull request merged**